src/test/interface/io/json/json_parser_test.cpp

```diff
@@ -363,231 +363,187 @@ TEST(ioJson,jsonParserErr01) {
 
 TEST(ioJson,jsonParserErr02) {
   test_exception("[ .5 ]",
-                 "illegal value, expecting object, array, number, string, or literal true/false/null\n");
+                 "Invalid value.\n");
 }
 
 TEST(ioJson,jsonParserErr02a) {
   test_exception("[ 0",
-                 "unexpected end of stream\n");
+                 "Missing a comma or ']' after an array element.\n");
 }
 
 TEST(ioJson,jsonParserErr02b) {
   test_exception("[ 0.",
-                 "unexpected end of stream\n");
+                 "Miss fraction part in number.\n");
 }
 
 
 TEST(ioJson,jsonParserErr02c) {
   test_exception("[ 99.9",
-                 "unexpected end of stream\n");
+                 "Missing a comma or ']' after an array element.\n");
 }
 
 
 TEST(ioJson,jsonParserErr03) {
   test_exception("[ 000.005 ]",
-                 "zero padded numbers not allowed\n");
+                 "Missing a comma or ']' after an array element.\n");
```

Comment:
if the error here is that 0-padded numbers are problematic, this message doesn't make sense.  are 0-padded numbers problematic?  can the parser handle '022'?

Reply:
Will get to the bottom of this, this one is weird.

Reply:
The reason for such a weird message is that the parser expects a comma or ] after the zero ("." is the third option if its a decimal number). 0-padded numbers are problematic and are not allowed per JSON standard. Some parsers do allow it. This one does not. One option of solving this would be to expand the error message to something like 
`"Missing a comma or ']' after an array element or found a zero padded number.\n"`

The other would be to change the guts of the parser.

Reply:
I added the above idea for now, we can revert it if its not fine.